nexus/src/sagas.rs

```diff
@@ -475,9 +522,121 @@ async fn sic_create_network_interfaces_undo(
     Ok(())
 }
 
+/// Create disks during instance creation, and return a list of disk names
+// TODO implement
+async fn sic_create_disks_for_instance(
+    sagactx: ActionContext<SagaInstanceCreate>,
+    disk_index: usize,
+) -> Result<Option<String>, ActionError> {
+    let saga_params = sagactx.saga_params();
+    let saga_disks = &saga_params.create_params.disks;
+
+    if disk_index >= saga_disks.len() {
+        return Ok(None);
+    }
+
+    let disk = &saga_disks[disk_index];
+
+    match disk {
+        params::InstanceDiskAttachment::Create(_create_params) => {
+            return Err(ActionError::action_failed(
+                "Creating disk during instance create unsupported!".to_string(),
+            ));
+        }
+
+        _ => {}
+    }
+
+    Ok(None)
+}
+
+/// Undo disks created during instance creation
+// TODO implement
+async fn sic_create_disks_for_instance_undo(
+    _sagactx: ActionContext<SagaInstanceCreate>,
+    _disk_index: usize,
+) -> Result<(), anyhow::Error> {
+    Ok(())
+}
+
+async fn sic_attach_disks_to_instance(
+    sagactx: ActionContext<SagaInstanceCreate>,
+    disk_index: usize,
+) -> Result<(), ActionError> {
+    ensure_instance_disk_attach_state(sagactx, disk_index, true).await
+}
+
+async fn sic_attach_disks_to_instance_undo(
+    sagactx: ActionContext<SagaInstanceCreate>,
+    disk_index: usize,
+) -> Result<(), anyhow::Error> {
+    Ok(ensure_instance_disk_attach_state(sagactx, disk_index, false).await?)
+}
+
+async fn ensure_instance_disk_attach_state(
+    sagactx: ActionContext<SagaInstanceCreate>,
+    disk_index: usize,
+    attached: bool,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let saga_params = sagactx.saga_params();
+    let opctx =
+        OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
+
+    let saga_disks = &saga_params.create_params.disks;
+    let instance_name = sagactx.lookup::<db::model::Name>("instance_name")?;
+
+    if disk_index >= saga_disks.len() {
+        return Ok(());
+    }
+
+    let disk = &saga_disks[disk_index];
+
+    let organization_name: db::model::Name =
+        saga_params.organization_name.clone().into();
+    let project_name: db::model::Name = saga_params.project_name.clone().into();
+
+    match disk {
+        params::InstanceDiskAttachment::Create(_) => {
+            // TODO grab disks created in sic_create_disks_for_instance
```

Comment:
Nit: If you wanna return an error here, we could (since this code should be uncallable right now)

Reply:
done in 959d49d92c885ddd94d739d77731b538dc40dc0b